### ISSUE TITLE
Newtonsoft old reference is being used in Insights project

### ISSUE THREAD

**Issue by UMW360** [Azure/azure-sdk-for-net#2871]
It looks like Azure Insights plugin still uses the older version of newtonsoft(6.0.8) which is not consistent with other plugins which are using newtonsoft (9.0.1). Is it possible to consolidate the newtonsoft reference across the Azure plugins

**Comment by nathandavidson** [comment]
If I'm understanding this issue correctly, my team is currently blocked by the same problem. The latest 0.15.0-preview version of the [Microsoft.Azure.Insights NuGet package](https://www.nuget.org/packages/Microsoft.Azure.Insights/) has a dependency on Newtonsoft.Json with a maximum version of 7.0.0, which means it cannot be installed along with other libraries in our project which depend on Newtonsoft.Json 9.0.0 or higher. The Newtonsoft.Json dependency version restrictions need to be relaxed so that Microsoft.Azure.Insights is compatible with current packages.

**Comment by johnwc** [comment]
There is also a issue with Microsoft.Rest.ClientRuntime referencing v6 for net45 but not the other two frameworks, they reference v9.  I created a [issue log](https://github.com/Azure/autorest/issues/2007) on their repo for this.

**Comment by Umamaheswaran1990** [comment]
@shahabhijeet Is there any other information you guys are looking for. I would be happy to provide any information that would help resolve this. As @nathandavidson pointed out it would be really helpful if we could relax the constraint

**Comment by AzureBinkie** [comment]
This problem is only getting worse as Newtonsoft.Json is now at version 10 and now no frameworks support the latest.

**Comment by svarcoe** [comment]
I agree with @AzureBinkie, the problem is getting worse. Seems like every Azure Management nuget package relies on a different version of Newtonsoft.Json.

**Comment by cleverguy25** [comment]
Can you change the package references to be a range, or accept all newtonsoft > 6?

**Comment by cleverguy25** [comment]
This is still tagged as MoreInfoFromCustomer, what extra info do you need?  This is blocking upgrade to newer azure client packages.

**Comment by cleverguy25** [comment]
@shahabhijeet no response?

**Comment by IvAlex1986** [comment]
Microsoft Team, please upgrade Newtonsoft.Json version for [Microsoft.Azure.Insights](https://www.nuget.org/packages/Microsoft.Azure.Insights/0.15.0-preview).
There are more than 200 nuget packages in our solution and restriction Newtonsoft.Json < 7.0.0 kills the Microsoft.Azure.Insights package for our solution.
Moreover, no one package uses Newtonsoft.Json version less than 7.0.0

**Comment by zvrba** [comment]
The proper solution is to abandon Newtonsoft.Json and use System.Runtime.Serialization.Json. No library has given me as much deployment troubles as Newtonsoft.Json. I've created a new ticket #4380 where I've added other links describing the situation.

**Comment by Petermarcu** [comment]
The Azure SDK is planning to remove references to JSON.NET in upcoming releases to avoid pinning dependencies like this. It will take time to get through all the different places where this dependency exists but over time, the pain should go away. We want people to be able to use whatever JSON.NET version they want without the Azure SDK causing problems.

You can get a glimpse of the first wave of libraries doing this here: https://azure.microsoft.com/en-in/blog/previewing-azure-sdks-following-new-azure-sdk-api-standards/
